Tools/CMake/modify_installed_headers.cmake

```diff
@@ -6,7 +6,27 @@ file(GLOB _installed_headers "${CMAKE_INSTALL_PREFIX}/include/*.H")
 list(REMOVE_ITEM _installed_headers "${CMAKE_INSTALL_PREFIX}/include/AMReX_Config.H")
 
 foreach( _header IN LISTS _installed_headers)
-   # This command only affect the first line of each file. Therefore, it can be used countless
-   # times on the same file and it always gives the same result (i.e. it does not keep prepending)
-   execute_process(COMMAND bash "-c" "sed -i '1s/^/#include <AMReX_Config.H>\\n\\n/' ${_header}" )
+  file(READ ${_header} _contents)
+  # Regex looks for:
+  # - Anything before include guard
+  # - #ifndef CPP_IDENTIFIER
+  # - #define CPP_IDENTIFIER
+  # - Rest of file
+  if (_contents MATCHES "^(.*)#ifndef +([A-Za-z0-9_]+)\n#define +([A-Za-z0-9_]+) *\n(.+)")
+    # We have found an include guard
+    if (${CMAKE_MATCH_2} STREQUAL ${CMAKE_MATCH_3})
+      # The guard defines match and are hence OK
+    else()
+      message(FATAL_ERROR "Include guard is wrong up for header file ${_header}")
+    endif()
+
+    # Insert the #include <AMReX_Config.H> between include guard and the rest of the file
+    string(CONCAT "_contents" "${CMAKE_MATCH_1}" "#ifndef " "${CMAKE_MATCH_2}" "\n#define " "${CMAKE_MATCH_3}" "\n#include <AMReX_Config.H>\n" "${CMAKE_MATCH_4}")
+
+  else()
+    message(WARNING "No include guard in header ${_header}")
+    string(PREPEND _contents "#include <AMReX_Config.H>\n")
```

Comment:
The warning and the include is unnecessary.  We have header files that are only included by other headers.

Reply:
I think there's only one `AMReX_FabArrayCommI.H`? Happy to remove the warning if you prefer of course

Reply:
There are a couple more in `Particles` and maybe other directories.  But I take my comment back.  It is probably better to add include guard to those files.